File: cat_ruby_object/lib/cat.rb

class Cat

  attr_accessor :mood, :hunger, :energy

  def initialize
    @mood = 50
    @hunger = 50
    @energy = 50
  end

  def feed
    @mood -= 10
    @hunger -= 20
    @energy += 10
    meow_if_mood
    meow_if_hunger
    meow_if_energy
  end

  def sleep
    @energy += 10
    meow_if_mood
    meow_if_hunger
    meow_if_energy
  end

  def play
    @mood += 10
    @hunger += 10
    @energy -= 20
    meow_if_mood
    meow_if_hunger
    meow_if_energy
  end

  def show_status
    puts "Mood : #{@mood}, Hunger : #{@hunger}, Energy : #{@energy} "
  end

  private

  def meow_if_hunger
     "Meow! feed me or play" if @hunger >= 80
  end

  def meow_if_mood
    "Meow! lets play" if @mood < 50
  end

  def meow_if_energy
    if @energy < 30
     "Meow! I need to sleep"
   else @energy > 30
     "Meow! get more energy"
   end
  end
end

File: cat_js_object/cat.js
class Cat{
    constructor(_mood, _hunger, _energy) {
    const mood = 50;
    const hunger = 50;
    const energy = 50;
  }

  feed() {
    this.mood -= 10
    this.hunger -= 20
    this.energy += 10
  }
  sleep() {
    this.energy += 10
  }

  play() {
    this.mood += 10
    this.hunger += 10
    this.energy -= 20
  }


}
